src/main/java/org/primefaces/extensions/component/fuzzysort/FuzzysortRenderer.java

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright 2011-2019 PrimeFaces Extensions
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.primefaces.extensions.component.fuzzysort;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.faces.component.UIComponent;
+import javax.faces.context.FacesContext;
+import javax.faces.context.ResponseWriter;
+import javax.faces.render.FacesRenderer;
+import org.primefaces.renderkit.CoreRenderer;
+import org.primefaces.util.WidgetBuilder;
+
+@FacesRenderer(componentFamily = Fuzzysort.COMPONENT_FAMILY, rendererType = Fuzzysort.DEFAULT_RENDERER)
+public class FuzzysortRenderer extends CoreRenderer {
+
+    @Override
+    public void encodeEnd(FacesContext context, UIComponent component) throws IOException {
+        if (context == null) {
+            throw new NullPointerException("No context defined!");
+        }
+
+        Fuzzysort fuzzysort = (Fuzzysort) component;
+
+        encodeMarkup(context, fuzzysort);
+        encodeScript(context, fuzzysort);
+    }
+
+    private void encodeScript(FacesContext context, Fuzzysort fuzzysort) throws IOException {
+        WidgetBuilder wb = getWidgetBuilder(context);
+
+        // TODO is it possible to make improvement here to have @FuzzyKey annotated class?
+        List list = (List) fuzzysort.getValue();
+        Object o = list.get(0);
+        Class<? extends Object> clazz = o.getClass();
+        Field[] fields = clazz.getDeclaredFields();
+        List<String> keys = Arrays.asList(fields).stream()
+                .filter(f -> f.getAnnotation(FuzzyKey.class) != null)
+                .map(f -> f.getName())
+                .collect(Collectors.toList());
+
```

Comment:
Maybe throw a FacesException here that says `"No @FuzzyKey annotation was detected on your class " + clazz + ". Please annotate one field in the class as @FuzzyKey.`